server/test/org/apache/cloudstack/outofbandmanagement/OutOfBandManagementServiceTest.java

```diff
@@ -0,0 +1,119 @@
+public class OutOfBandManagementServiceTest {
+
+    OutOfBandManagementServiceImpl oobmService = new OutOfBandManagementServiceImpl();
+
+    @Test
+    public void testOutOfBandManagementDriverResponseEvent() {
+        OutOfBandManagementDriverResponse r = new OutOfBandManagementDriverResponse("some result", "some error", false);
+
+        r.setSuccess(false);
+        r.setAuthFailure(false);
+        Assert.assertEquals(r.toEvent(), OutOfBandManagement.PowerState.Event.Unknown);
+
+        r.setSuccess(false);
+        r.setAuthFailure(true);
+        Assert.assertEquals(r.toEvent(), OutOfBandManagement.PowerState.Event.AuthError);
+
+        r.setAuthFailure(false);
+        r.setSuccess(true);
+        r.setPowerState(OutOfBandManagement.PowerState.On);
+        Assert.assertEquals(r.toEvent(), OutOfBandManagement.PowerState.Event.On);
+
+        r.setPowerState(OutOfBandManagement.PowerState.Off);
+        Assert.assertEquals(r.toEvent(), OutOfBandManagement.PowerState.Event.Off);
+
+        r.setPowerState(OutOfBandManagement.PowerState.Disabled);
+        Assert.assertEquals(r.toEvent(), OutOfBandManagement.PowerState.Event.Disabled);
+    }
+
+    private ImmutableMap<OutOfBandManagement.Option, String> buildRandomOptionsMap() {
+        ImmutableMap.Builder<OutOfBandManagement.Option, String> builder = new ImmutableMap.Builder<>();
+        builder.put(OutOfBandManagement.Option.ADDRESS, "localhost");
+        builder.put(OutOfBandManagement.Option.DRIVER, "ipmitool");
+        return builder.build();
+    }
+
+    @Test
+    public void testUpdateOutOfBandManagementConfigValid() {
+        OutOfBandManagement config = new OutOfBandManagementVO(123L);
+        Assert.assertEquals(config.getPowerState(), OutOfBandManagement.PowerState.Disabled);
+        config = oobmService.updateConfig(config, buildRandomOptionsMap());
+        Assert.assertEquals(config.getAddress(), "localhost");
+        Assert.assertEquals(config.getDriver(), "ipmitool");
+        Assert.assertEquals(config.getPowerState(), OutOfBandManagement.PowerState.Disabled);
+    }
+
+    @Test
+    public void testUpdateOutOfBandManagementConfigInValid() {
+        try {
+            oobmService.updateConfig(null, buildRandomOptionsMap());
+            Assert.fail("CloudRuntimeException was expect for out-of-band management not configured for the host");
+        } catch (CloudRuntimeException ignored) {
+        }
+
+        try {
+            oobmService.updateConfig(null, null);
+            Assert.fail("CloudRuntimeException was expect for out-of-band management not configured for the host");
+        } catch (CloudRuntimeException ignored) {
+        }
+
+        OutOfBandManagement config = new OutOfBandManagementVO(123L);
+        config.setAddress(null);
+        config = oobmService.updateConfig(config, null);
+        Assert.assertEquals(config.getAddress(), null);
+        Assert.assertEquals(config.getPowerState(), OutOfBandManagement.PowerState.Disabled);
```

Comment:
Consider dividing this test method into three -- one for each failure scenario and another for the success scenario.  Also, for the failure methods, use the ``expected`` attribute on the ``@Test`` annotation.